pkg/apis/pipeline/v1alpha2/pipeline_types.go

```diff
@@ -227,3 +232,29 @@ type PipelineList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Pipeline `json:"items"`
 }
+
+// PipelineTaskRunOn represents runOn of a pipelineTask which is a map of task and a list of states, e.g.
+// runOn:
+// 	- task: task1
+//    states: ["success", "failure"]
+//  - task: task2
+//    states: ["success"]
+type PipelineTaskRunOn struct {
+	// Task is the name of the PipelineTask as specified in a pipeline
+	Task string `json:"task"`
+	// States is a list of states of a PipelineTask which is specified in the Task above
+	States []PipelineTaskState `json:"states"`
+}
```

Comment:
`runOn` is a map with key as a task name and value as a list of states

Reply:
might be a good time to bikeshed on the `runOn` format i.e. map vs list. And I think @skaegi has some ideas around here as well :)